wordpress-mobile/WordPress-Android
# [Jetpack Social] Implement customize share message screen

~Should be merged after https://github.com/wordpress-mobile/WordPress-Android/pull/18784~

Should be merged after (and also after updating the FluxC version) https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2777

Part of #18571 

Unit tests for the new screen will be added in my next PR.

<p float="left">
<img width="300" alt="Screen Shot 2023-07-19 at 6 58 00 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/99be8799-b647-4164-8e55-e1e485fd1fdf">
<img width="300" alt="Screen Shot 2023-07-19 at 6 57 11 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/9f59c044-1c6a-4b07-aaa8-f6227fde9294">
</p>

Since this is a new screen for Android, I've tried to match the design that already exists in iOS:

<p float="left">
<img width="300" alt="Screen Shot 2023-07-19 at 7 02 11 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/6d092ac9-4967-4d3b-af86-b7c8be187f4c">
<img width="300" alt="Screen Shot 2023-07-19 at 7 01 39 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/d684261a-50b8-405c-bc6d-4160889faa07">
</p>

Besides implementing the same elements that iOS already have, I've tried to match the same style of tags screen on Android:

<p float="left">
<img width="300" alt="Screen Shot 2023-07-19 at 7 02 32 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/fe5de838-5926-4327-8aee-327f6eedab3b">
<img width="300" alt="Screen Shot 2023-07-19 at 7 02 57 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/5e729a6f-5ba7-4a5e-843f-fad5e8a534e4">
</p>

To test:
1 - Enable `JetpackSocialFeatureConfig` feature flag;
2 - Select a site that has at least one social sharing connection;
3 - Open post settings of an unpublished post: jetpack social section should be shown;
5 - Tap the message container to open the customize message screen;
6 - Delete all the text and move back to post settings screen: the previously selected message should be shown;
7 - Tap the message container again: change the message and move back. The message should be updated in post settings message container. Do the same while also changing the device orientation;
8 - Tap the back button to move back to the post editor, and then back button again to return to the posts list: the draft should be uploaded and the share message should be synced to back-end. To make sure it's synced, open post settings on web/iOS;
9 - Change the message in web/iOS and open it on Android to make sure it's updated properly;
10 - Disable `JetpackSocialFeatureConfig` feature flag;
11 - Open post settings screen again: jetpack social section should NOT be shown.

If you open a post that's already published, it's going to be possible to edit the share message in the client, but this is going to change. In another PR I'm going to implement the enable/disable business rules of Jetpack Social section in Post Settings. This container be disabled if the post is already published (and also considering the number of remaining shares).

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and unit testing.

3. What automated tests I added (or what prevented me from doing so)
Updated existing unit tests and new unit tests will be added in my next PR.

PR submission checklist:


UI Changes testing checklist:
